examples/topography_spectra.jl

```diff
@@ -0,0 +1,133 @@
+using LazyArtifacts
+using ClimaAtmos.AtmosArtifacts
+import ClimaAtmos as CA
+import ClimaCore as CC
+import ClimaComms
+import ClimaParams as CP
+import ClimaCore.Quadratures
+using ClimaUtilities: SpaceVaryingInputs
+using ClimaCore.Hypsography: diffuse_surface_elevation!
+using ClimaCore: Remapping, Geometry, Operators, Spaces
+using Plots, ClimaCorePlots
+
+import ClimaCoreSpectra: power_spectrum_2d
+
+const AA = AtmosArtifacts
+
+"""
+    generate_spaces(; h_elem=60, diffiter=32)
+For a given number of elements `h_elem`, this function generates a 
+horizontal spectral space (ClimaCore.SpectralElementSpace2D) and
+generates an interpolated (using SpaceVaryingInputs) orography field
+on this space. The `diffuse_surface_elevation` function is called 
+to apply `diffiter` instances of Laplacian diffusion passes 
+(which rely on the canonical unsteady diffusion equation).
+Returns a `Field` containing surface orography. 
+"""
+function generate_spaces(; h_elem = 60, diffiter = 32)
+    FT = Float32
+    param_dict = CP.create_toml_dict(FT)
+    params = CP.get_parameter_values(param_dict, ["planet_radius"])
+    cubed_sphere_mesh =
+        CA.cubed_sphere_mesh(; radius = params.planet_radius, h_elem)
+    quad = Quadratures.GLL{4}()
+    comms_ctx = ClimaComms.SingletonCommsContext()
```

Comment:
Why is this restriction?

Reply:
This example is light enough that it can be run on a single device, so MPI / generalized context was unnecessary. I'll generalize with `ClimaComms.context()`.